### PR TITLE
[generator] Deriving from certain protocol interfaces generates uncompilable code. Fixes bug 52665

### PR DESCRIPTION
**Pull request by dalexsoto**
Some namespaces are missing on generated files, causing error CS0246
The type or namespace name `Foo` could not be found.

While fixing this another error arose, by enabling the following test

`[Protocol] interface C140 : ICIImageProcessorInput {}`

some code paths lead to CVPixelBuffer creation which our code
tries to invoke the handle ctor and fails to find it because it
is internal. This will be fixed in a different commit.

build diff: https://gist.github.com/dalexsoto/055381e7aebecca254139bd6cca6a2e8

### CODE REVIEW

File: src/generator.cs [hunk 1]
```diff
@@ -717,6 +717,19 @@ public NamespaceManager (string prefix, string customObjCRuntimeNS, bool skipSys
 		if (Frameworks.HaveMetal && !(Generator.CurrentPlatform == PlatformName.MacOSX && !Generator.UnifiedAPI))
 			ImplicitNamespaces.Add (Get ("Metal"));
 
+		if (Frameworks.HaveCoreImage && !(Generator.CurrentPlatform == PlatformName.WatchOS))
```
Comment (rolfbjarne) [inline src/generator.cs]: Why do you need the additional platform checks?

In the code above they're only there to not change the generated output when I ported the code to use IKVM; in theory those checks can be removed (but when porting to IKVM I preferred fixing one thing at a time, minimizing the diff in the generated output).

Comment (spouliot) [inline src/generator.cs]: Good point, It would be cleaner/simpler without them 👍

**Comment by dalexsoto** [comment]
@rolfbjarne / @spouliot Fixed 👍

**Comment by dalexsoto** [comment]
@spouliot noted, will do!

**Pull request merged**